mmv1/products/networkconnectivity/PolicyBasedRoute.yaml

```diff
@@ -0,0 +1,194 @@
+    name: 'name'
+    required: true
+    url_param_only: true
+    description: |
+      The name of the policy based route.
+  - !ruby/object:Api::Type::String
+    name: 'description'
+    description: |
+      An optional description of this resource.
+  - !ruby/object:Api::Type::KeyValueLabels
+    name: 'labels'
+    description: |
+      User-defined labels.
+  - !ruby/object:Api::Type::String
+    name: 'network'
+    required: true
+    description: |
+      Fully-qualified URL of the network that this route applies to, for example: projects/my-project/global/networks/my-network.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'filter'
+    required: true
+    description: |
+      The filter to match L4 traffic.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'protocolVersion'
+        required: true
+        description: |
+          Internet protocol versions this policy-based route applies to.
+        values:
+          - :IPV4
+      - !ruby/object:Api::Type::String  # probably could have been an enum, but it's a string in the API
+        name: 'ipProtocol'
+        default_value: "ALL"
+        description: |
+          The IP protocol that this policy-based route applies to. Valid values are 'TCP', 'UDP', and 'ALL'. Default is 'ALL'.
+      - !ruby/object:Api::Type::String
+        name: 'srcRange'
+        default_value: "0.0.0.0/0"
+        description: |
+          The source IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+      - !ruby/object:Api::Type::String
+        name: 'destRange'
+        default_value: "0.0.0.0/0"
+        description: |
+          The destination IP range of outgoing packets that this policy-based route applies to. Default is "0.0.0.0/0" if protocol version is IPv4.
+  - !ruby/object:Api::Type::Enum
+    name: 'nextHopOtherRoutes'
+    exactly_one_of:  # next_hop union
+      - next_hop_ilb_ip
+      - next_hop_other_routes
+    description: |
+      Other routes that will be referenced to determine the next hop of the packet.
+    values:
+      - :DEFAULT_ROUTING
+  - !ruby/object:Api::Type::String
+    name: 'nextHopIlbIp'
+    exactly_one_of:  # next_hop union
+      - next_hop_ilb_ip
+      - next_hop_other_routes
+    description: |
+      The IP address of a global-access-enabled L4 ILB that is the next hop for matching packets.
+  - !ruby/object:Api::Type::Integer
+    name: 'priority'
+    default_value: 1000
+    description: |
+      The priority of this policy-based route. Priority is used to break ties in cases where there are more than one matching policy-based routes found. In cases where multiple policy-based routes are matched, the one with the lowest-numbered priority value wins. The default value is 1000. The priority value must be from 1 to 65535, inclusive.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'virtualMachine'
+    exactly_one_of:  # target union
```

Comment:
same question for the `target` union field.

Reply:
No, good catch! One of my tests is gonna fail :(

I added this union after writing the basic test so hadn't run it locally